internal/telemetry/cluster.go

```diff
@@ -35,3 +37,81 @@ func (c *Collector) K8sVersion() (string, error) {
 	}
 	return sv.String(), nil
 }
+
+// Platform returns a string representing platform name.
+func (c *Collector) Platform(ctx context.Context) (string, error) {
+	nodes, err := c.Config.K8sClientReader.CoreV1().Nodes().List(ctx, metaV1.ListOptions{})
+	if err != nil {
+		return "", err
+	}
+	if len(nodes.Items) == 0 {
+		return "", errors.New("no nodes in the cluster, cannot determine platform name")
+	}
+	return lookupPlatform(nodes.Items[0].Spec.ProviderID), nil
+}
+
+// lookupPlatform takes a string representing a K8s PlatformID
+// retrieved from a cluster node and returns a string
+// representing the platform name.
+//
+// Cloud providers identified by PlatformID (in K8s SIGs):
+// https://github.com/orgs/kubernetes-sigs/repositories?q=cluster-api-provider
+//
+//gocyclo:ignore
+func lookupPlatform(providerID string) string {
+	provider := strings.TrimSpace(providerID)
+	// The case when the ProviderID field not used by the cloud provider.
+	if provider == "" {
+		return "other"
+	}
+
+	provider = strings.ToLower(providerID)
```

Comment:
I feel like we can simplify this function by putting the platforms in a map and iterating that map.

For example:
```go
// key = providerId; value = platform
var mapProviderIDToPlatform = map[string]string{
	"aws":          "aws",
	"azure":        "azure",
	"gce":          "gke",
	"kind":         "kind",
	"vsphere":      "vsphere",
	"k3s":          "k3s",
	"ibmcloud":     "ibmcloud",
	"ibmpowervs":   "ibmpowervs",
	"cloudstack":   "cloudstack",
	"openstack":    "openstack",
	"digitalocean": "digitalocean",
	"equinixmetal": "equinixmetal",
	"alicloud":     "alicloud",
}
```

```go
func lookupPlatform(providerID string) string {
	provider := strings.TrimSpace(providerID)
	// The case when the ProviderID field not used by the cloud provider.
	if provider == "" {
		return "other"
	}

	provider = strings.ToLower(providerID)

	for id, platform := range mapProviderIDToPlatform {
		if strings.HasPrefix(provider, id) {
			return platform
		}
	}
}
```

Reply:
There was a comment @pleshakov made on [here](https://github.com/nginxinc/nginx-gateway-fabric/pull/1651#discussion_r1516245954) about defining a type for constants, and then mapping to those constants [here](https://github.com/nginxinc/nginx-gateway-fabric/pull/1651#discussion_r1516250232)

What are your thoughts on this? I'm in the same boat as @bjee19. I don't fully see the benefit of setting `type CusterPlatform string`